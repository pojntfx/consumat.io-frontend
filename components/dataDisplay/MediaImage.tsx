import { PhotographIcon } from "@heroicons/react/outline";
import { useEffect, useState } from "react";

type ImageProps = {
  imageSrc: string;
  className?: string;
};

const MediaImage = ({ imageSrc, className }: ImageProps) => {
  const [isError, setIsError] = useState(false);
  const [isEmpty, setIsEmpty] = useState(false);

  useEffect(() => {
    if (imageSrc == null) {
      setIsEmpty(true);
    }
  }, [imageSrc]);

  return (
    <div className={className + " flex-shrink-0"}>
      {!isError && !isEmpty ? (
        <img
          className="h-full w-full object-cover"
          src={imageSrc}
          onError={(event) => {
            setIsError(true);
          }}
        />
      ) : (
        <div className="h-full w-full flex items-center justify-center bg-gray-200">
          <PhotographIcon className="h-10 w-10 text-gray-300" />
        </div>
      )}
    </div>
  );
};

export default MediaImage;
