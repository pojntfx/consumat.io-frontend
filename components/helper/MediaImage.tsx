import { PhotographIcon } from "@heroicons/react/outline";
import { useState } from "react";

type ImageProps = {
  imageSrc: string;
  className?: string;
};

const MediaImage = ({ imageSrc, className }: ImageProps) => {
  const [isError, setIsError] = useState(false);

  return (
    <div className={className + " flex-shrink-0"}>
      {!isError ? (
        <img
          className="h-full w-full object-cover"
          src={imageSrc}
          onError={() => {
            setIsError(true);
          }}
        />
      ) : (
        <div className="h-full w-full flex items-center justify-center bg-gray-200">
          <PhotographIcon className="h-10 w-10 text-gray-300" />
        </div>
      )}
    </div>
  );
};

export default MediaImage;
