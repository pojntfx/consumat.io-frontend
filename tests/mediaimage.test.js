import React from "react";
import renderer from "react-test-renderer";
import MediaImage from "../components/dataDisplay/MediaImage";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MediaImage
        imageSrc={testExamples.exampleImgSrc}
        className={testExamples.exampleClassname}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
