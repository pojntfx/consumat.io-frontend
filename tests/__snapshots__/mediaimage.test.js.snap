// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ml-2 flex-shrink-0"
>
  <img
    className="h-full w-full object-cover"
    onError={[Function]}
    src="https://image.tmdb.org/t/p/w154/ibiGOAzumOIzCk8fHwv03jzYbj9.jpg"
  />
</div>
`;
